Add Setting color interaction states (Filled, Hover, Active), wire for ExternalLink and ValueInput Settings

jarolrod opened the pull request:
Introduces the Filled, Hover and Active states for Settings. Will introduce the rest of the states in follow-ups.

light
| Filled | Hover | Active |
| ------ | ----- | ------ |
| <img width="454" alt="Screen Shot 2023-02-01 at 1 42 17 AM" src="https://user-images.githubusercontent.com/23396902/215970965-c3b646dc-f70a-41dd-80e9-add8e461bad3.png"> | <img width="454" alt="Screen Shot 2023-02-01 at 1 42 26 AM" src="https://user-images.githubusercontent.com/23396902/215970984-c7a8aa8e-d23d-40a1-9e7f-aec0709c536d.png"> | <img width="454" alt="Screen Shot 2023-02-01 at 1 42 34 AM" src="https://user-images.githubusercontent.com/23396902/215971011-9cd829a5-c643-4bb0-8666-e22d25214a97.png"> |

dark 
| Filled | Hover | Active |
| ------ | ----- | ------ |
| <img width="454" alt="Screen Shot 2023-02-01 at 1 41 29 AM" src="https://user-images.githubusercontent.com/23396902/215970689-bd23ca8e-7234-4a37-a9bc-20de8f72049a.png"> | <img width="454" alt="Screen Shot 2023-02-01 at 1 41 43 AM" src="https://user-images.githubusercontent.com/23396902/215970732-38806d33-a2b8-46f0-a9fa-733030a2e62a.png"> | <img width="454" alt="Screen Shot 2023-02-01 at 1 41 51 AM" src="https://user-images.githubusercontent.com/23396902/215970768-e391c2af-0c21-47e7-a541-89f1e76a0e2f.png"> |







[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/<PR>)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/<PR>)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/<PR>)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/<PR>)

johnny9 reviewed (commented):
Concept ACK

Code looks good, will just need to rebase with main once #240 is merged and can re-review and approve. It looks like it might work with a rebase to main now, though.

hebasto commented:
> Based on #240

It has been just merged. Rebase?

jarolrod commented:
updated from[ bb997ec ](https://github.com/bitcoin-core/gui-qml/commit/bb997ecce3130eca9b4d7f0ee46cb3c540109838)to [970c66f](https://github.com/bitcoin-core/gui-qml/commit/970c66f3e8a631bfd7484138a0e1dee424210b77), [compare](https://github.com/bitcoin-core/gui-qml/compare/bb997ecce3130eca9b4d7f0ee46cb3c540109838..970c66f3e8a631bfd7484138a0e1dee424210b77)

changes: rebased over master

jarolrod commented:
cc @johnny9

johnny9 reviewed (approved):
ACK 970c66f3e8a631bfd7484138a0e1dee424210b77
